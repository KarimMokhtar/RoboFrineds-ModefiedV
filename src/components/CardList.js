import React from "react";
import Card from "./Card";

const CardList = ({ robots }) => {
  return (
    <div>
      {robots.map((robot) => (
        <Card email={robot.email} name={robot.name} key={robot.id} />
      ))}
    </div>
  );
};

export default CardList;
