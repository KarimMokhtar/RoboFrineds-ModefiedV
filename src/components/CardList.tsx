import React from "react";
import Card from "./Card";

interface ICardList {
  robots:Array<any>
}
const CardList = ({ robots }:ICardList) => {
  return (
    <div>
      {robots.map((robot) => (
        <Card email={robot.email} name={robot.name} key={robot.id} />
      ))}
    </div>
  );
};

export default CardList;
