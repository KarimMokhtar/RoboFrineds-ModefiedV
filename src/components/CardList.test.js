import React from "react";

import { shallow } from "enzyme";
import CardList from "./CardList";
it("tests the card list component", () => {
  const mockRobots = [
    {
      id: 1,
      name: "test1",
      email: "test1@rad.com",
    },
  ];
  const wrapper = shallow(<CardList robots={mockRobots} />);
  expect(wrapper.debug()).toMatchSnapshot();
});
