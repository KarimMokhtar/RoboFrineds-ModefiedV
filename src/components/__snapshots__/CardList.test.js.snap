// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the card list component 1`] = `
"<div>
  <Card email=\\"test1@rad.com\\" name=\\"test1\\" />
</div>"
`;
